cmd/server/search.go

```diff
@@ -213,12 +223,19 @@ func (s *searcher) TopAltNames(limit int, alt string) []Alt {
 	}
 	xs := newLargest(limit)
 
+	var wg sync.WaitGroup
+	wg.Add(len(s.Alts))
```

Comment:
Should we add some kind of limit to avoid kicking off unknown quantities of goroutines?

Reply:
Yea and it is sorta handled by default. Goroutines are multiplexed across the available CPU's on the machine. This means they're not individual threads (in Java's case) and are a lot lighter weight.

Reply:
Typically `runtime.GOMAXPROCS(runtime.NUMCPU())` is used to limit this to the available CPU's, but I'm pretty sure that's the default now. 

https://github.com/letsencrypt/boulder/blob/5457680a9c8ce34d0456ccf289ed347a8529a31e/cmd/cert-checker/main.go#L335 is a common pattern to use

Reply:
Oh cool. I didn't know about the baked in limiting.